unit-tests/src/test/scala/scala/scalanative/unsafe/PtrOpsSuite.scala

```diff
@@ -24,4 +24,23 @@ object PtrOpsSuite extends tests.Suite {
       assert(sarr - sptr == -7)
     }
   }
+
+  test("cast to CFuncPtr") {
+    type ExpectedFnType = CFuncPtr0[CInt]
+    type OtherFnType    = CFuncPtr1[CInt, CString]
+    val funcPtr: Ptr[Byte] = new ExpectedFnType {
+      override def apply(): CInt = stdlib.rand()
+    }.toPtr
+
+    assertTrue {
+      Ptr
+        .ptrToCFuncPtr[ExpectedFnType](funcPtr)
+        .isInstanceOf[ExpectedFnType]
```

Comment:
Same comment about the meaning of the test. We should prevent the implementation from returning a dummy function or something like that.